file_writer_control/WorkerStatus.py

```diff
@@ -0,0 +1,70 @@
+from datetime import datetime
+from enum import Enum, auto
+
+
+class WorkerState(Enum):
+    """
+    The state of a worker (i.e. a file-writer instance).
+    """
+    IDLE = auto()
+    WRITING = auto()
+    UNKNOWN = auto()
+    UNAVAILABLE = auto()
+
+
+class WorkerStatus(object):
+    """
+    Contains general status information about a worker.
+    """
+    def __init__(self, service_id: str):
+        self._last_update = datetime.now()
+        self._service_id = service_id
+        self._state = WorkerState.UNAVAILABLE
+
+    def __eq__(self, other_status: 'WorkerStatus') -> bool:
+        if not isinstance(other_status, WorkerStatus):
+            raise NotImplementedError
+        return self.service_id == other_status.service_id and self.state == other_status.state
+
+    def update_status(self, new_status: 'WorkerStatus'):
+        """
+        Updates the status/state of this instance of the WorkerStatus class using another instance.
```

Comment:
May be we can use `` to highlight class and/or function names in the docstrings. There similar cases in the code below. I do not know if I missed this in the previous PRs. Not very important though.

Reply:
Hm, neither of two Python editors does highlighting when surrounding a name with `` so I am not sure it is worth it. I can still do it if you insist but I prefer not too.

Reply:
No, there is nothing to insists. If or when we crate documentation, we can revisit. Otherwise, not so important.

Reply:
Just to clarify, when we use ``, it highlights these in the API docs not in the editors.